### PR TITLE
Persist stableIDs across subdomains on StatsigReporter.js

### PR DESCRIPTION
**Pull request by kelbyhawn**
Sets up a function to send a persistent stableID to Statsig for experiments. 

## Links
Jira ticket: [ACQ-2074](https://codedotorg.atlassian.net/browse/ACQ-2074)

## Testing story
Tested locally and confirmed the same stableID's were showing up across code.org, studio.code.org, and in Statsig. 

----

## code.org
Screenshot 2024-07-23 at 2 26 56 PM

## studio.code.org
Screenshot 2024-07-23 at 2 27 33 PM

## Statsig
Screenshot 2024-07-23 at 2 30 38 PM

### CODE REVIEW

File: apps/src/lib/util/StatsigReporter.js [hunk 1]
```diff
@@ -125,6 +129,19 @@ class StatsigReporter {
     }
   }
 
+  findOrCreateStableId() {
+    const stableIdKey = 'statsig_stable_id';
+    let stableId = cookies.get(stableIdKey);
+    if (!stableId) {
+      stableId = createUuid();
+      cookies.set(stableIdKey, stableId, {
+        expires: 400,
```
Comment (kelbyhawn) [inline apps/src/lib/util/StatsigReporter.js:138]: Cookie will expire in 400 days. (Note that this is in a new incognito browser so this screenshot won't have the same stableID as the other screenshots.)

Screenshot 2024-07-24 at 9 35 30 AM

File: apps/src/lib/util/StatsigReporter.js [hunk 2]
```diff
@@ -125,6 +129,19 @@ class StatsigReporter {
     }
   }
 
+  findOrCreateStableId() {
+    const stableIdKey = 'statsig_stable_id';
```
Comment (TurnerRiley) [inline apps/src/lib/util/StatsigReporter.js]: Nit: If this is a string constant, can it be defined up by the other constants iin lines 15-16 please

File: apps/src/lib/util/StatsigReporter.js [hunk 3]
```diff
@@ -7,6 +8,7 @@ import {
   getEnvironment,
   isProductionEnvironment,
   isDevelopmentEnvironment,
+  createUuid,
```
Comment (TurnerRiley) [inline apps/src/lib/util/StatsigReporter.js:11]: Nit: What does the second "u" stand for? Should it be "createUid"?

**Review by TurnerRiley — APPROVED** [review]
LGTM! Just had some tiny nits

File: apps/src/lib/util/StatsigReporter.js (hunk 3 above)
Comment (bethanyaconnor) [reply apps/src/lib/util/StatsigReporter.js:11]: UUID stands for "Universally Unique Identifier".

**Review by bethanyaconnor — APPROVED** [review]
Looks good after addressing Turner's comment!

File: apps/src/lib/util/StatsigReporter.js (hunk 3 above)
Comment (TurnerRiley) [reply apps/src/lib/util/StatsigReporter.js:11]: Ah thank you!

File: apps/src/lib/util/StatsigReporter.js (hunk 2 above)
Comment (kelbyhawn) [reply apps/src/lib/util/StatsigReporter.js]: Yes def!

**Pull request merged**